Mathlib/Analysis/Calculus/ContDiff/Defs.lean

```diff
@@ -381,16 +397,19 @@ theorem hasFTaylorSeriesUpToOn_succ_iff_right {n : ℕ} :
       · have A : (m : ℕ∞) < n := by
           rw [Nat.cast_lt] at hm ⊢
           exact Nat.lt_of_succ_lt_succ hm
-        have :
-          HasFDerivWithinAt ((continuousMultilinearCurryRightEquiv' 𝕜 m E F).symm ∘ (p · m.succ))
-            ((p x).shift m.succ).curryLeft s x := Htaylor.fderivWithin _ A x hx
-        rw [LinearIsometryEquiv.comp_hasFDerivWithinAt_iff'] at this
-        convert this
-        ext y v
-        change
+        have eq : (p x (m + 1).succ).curryLeft =
+            (continuousMultilinearCurryRightEquiv' 𝕜 m E F : ContinuousLinearMap ..).comp
+            ((p x).shift (m + 1)).curryLeft := by
+          ext y v
+          change
           (p x (Nat.succ (Nat.succ m))) (cons y v) =
             (p x m.succ.succ) (snoc (cons y (init v)) (v (last _)))
-        rw [← cons_snoc_eq_snoc_cons, snoc_init_self]
+          rw [← cons_snoc_eq_snoc_cons, snoc_init_self]
+        rw [eq]
+        have :
+          HasFDerivWithinAt ((continuousMultilinearCurryRightEquiv' 𝕜 m E F).symm ∘ (p · m.succ))
+            ((p x).shift m.succ).curryLeft s x := by apply Htaylor.fderivWithin _ A x hx
+        rwa [LinearIsometryEquiv.comp_hasFDerivWithinAt_iff'] at this
```

Comment:
Is this change (adding three lines of code) preferable to just adding `set_option maxSynthPendingDepth 2 in` above the theorem? On my machine, the maxSynthPendingDepth 2 option in conjunction with switching the profiler on uses 51K heartbeats and 1.9 seconds, whereas the fix in this PR uses 78K heartbeats and 2.8 seconds. Should `set_option maxSynthPendingDepth 2` be thought of as a code smell?

Reply:
Using set_option.